src/styles/components/_component-card.scss

```diff
@@ -1,20 +1,53 @@
-.component-link {
-  display: block;
-  outline: none;
+.component-card {
+  display: flex;
+  flex-direction: column-reverse;
+  height: 21.875rem; // 350px/16
   width: 100%;
-  height: 100%;
+  cursor: pointer;
+  transition: all 150ms ease-in-out;
+  border: 2px solid var(--foreground-color);
+  background-color: var(--background-color);
 
-  .component-card {
-    display: flex;
-    flex-direction: column-reverse;
-    height: 21.875rem; // 350px/16
+  &__image {
+    box-sizing: border-box;
+    transition: all 150ms ease-in-out;
     width: 100%;
-    cursor: pointer;
+    height: 70%;
+    padding: min(15%, 2rem);
+    margin: auto;
+  }
+
+  @media (prefers-color-scheme: dark) {
+    filter: invert(1);
+  }
+
+  @media (min-width: $breakpoint-scale-up) {
+    height: 100%;
+  }
```

Comment:
I would nest this with `.component-card` to make sure the cards fill the `.component-link` fully.